lang/test/org/partiql/lang/eval/EvaluatingCompilerNAryTests.kt

```diff
@@ -46,78 +43,88 @@ class EvaluatingCompilerNAryTests: EvaluatorTestBase() {
+        ArithmeticTestCase("minus", 10, 1, null, null),
 
         //SUB is noncommutative
-        ArithmeticTestCase(NAryOp.SUB, 10, 1, 2, 7),
-        ArithmeticTestCase(NAryOp.SUB, 1, 2, 10, -11),
+        ArithmeticTestCase("minus", 10, 1, 2, 7),
+        ArithmeticTestCase("minus", 1, 2, 10, -11),
 
         //Null propagation for MUL
-        ArithmeticTestCase(NAryOp.MUL, null, 2, 3, null),
-        ArithmeticTestCase(NAryOp.MUL, 10, null, 3, null),
-        ArithmeticTestCase(NAryOp.MUL, 10, 2, null, null),
+        ArithmeticTestCase("times", null, 2, 3, null),
+        ArithmeticTestCase("times", 10, null, 3, null),
+        ArithmeticTestCase("times", 10, 2, null, null),
 
         //MUL is commutative
-        ArithmeticTestCase(NAryOp.MUL, 10, 2, 3, 60),
-        ArithmeticTestCase(NAryOp.MUL, 2, 3, 10, 60),
+        ArithmeticTestCase("times", 10, 2, 3, 60),
+        ArithmeticTestCase("times", 2, 3, 10, 60),
 
         //Null propagation for DIV
-        ArithmeticTestCase(NAryOp.DIV, null, 2, 3, null),
-        ArithmeticTestCase(NAryOp.DIV, 10, null, 3, null),
-        ArithmeticTestCase(NAryOp.DIV, 10, 2, null, null),
+        ArithmeticTestCase("divide", null, 2, 3, null),
+        ArithmeticTestCase("divide", 10, null, 3, null),
+        ArithmeticTestCase("divide", 10, 2, null, null),
 
         //DIV is noncommutative
-        ArithmeticTestCase(NAryOp.DIV, 60, 2, 3, 10),
-        ArithmeticTestCase(NAryOp.DIV, 2, 3, 10, 0),
+        ArithmeticTestCase("divide", 60, 2, 3, 10),
+        ArithmeticTestCase("divide", 2, 3, 10, 0),
 
         //Null propagation for MOD
-        ArithmeticTestCase(NAryOp.MOD, null, 2, 3, null),
-        ArithmeticTestCase(NAryOp.MOD, 10, null, 3, null),
-        ArithmeticTestCase(NAryOp.MOD, 10, 2, null, null),
+        ArithmeticTestCase("modulo", null, 2, 3, null),
+        ArithmeticTestCase("modulo", 10, null, 3, null),
+        ArithmeticTestCase("modulo", 10, 2, null, null),
 
         //MOD is noncommutative
-        ArithmeticTestCase(NAryOp.MOD, 19, 5, 3, 1),
-        ArithmeticTestCase(NAryOp.MOD, 5, 3, 19, 2)
+        ArithmeticTestCase("modulo", 19, 5, 3, 1),
+        ArithmeticTestCase("modulo", 5, 3, 19, 2)
     )
 
     @Test
     @Parameters
     fun ternaryArithmeticTest(tc: ArithmeticTestCase) {
-        val exprNode = NAry(
-            tc.op,
-            listOf(
-                Literal(tc.arg1.toIonValue(), dummyMetas),
-                Literal(tc.arg2.toIonValue(), dummyMetas),
-                Literal(tc.arg3.toIonValue(), dummyMetas)
-            ), dummyMetas)
+        fun buildExprList(arg1: Long?, arg2: Long?, arg3: Long?) = listOf(
+            PartiqlAst.build { lit(arg1.toIonValue().toIonElement()) },
+            PartiqlAst.build { lit(arg2.toIonValue().toIonElement()) },
+            PartiqlAst.build { lit(arg3.toIonValue().toIonElement()) },
+        )
 
+        val query = PartiqlAst.build {
+            query(
+                when (tc.op) {
```

Comment:
Blah, I'd consider making `tc.op` an enum--don't use strings for this.

Reply:
Will create enum for these operators in the next commit.